library/insights_registration.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Terry Jones <terry.jones@example.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: insights_registration
+
+short_description: This module registers the insights client
+
+description:
+    - This module will check the current registration status, unregister if needed,
+    and then register the insights client (and update the display_name if needed)
+
+options:
+    insights_name:
```

Comment:
Same as earlier, can this be optional?  And default to the correct current behavior of the client?